test/functional/wallet_sendall.py

```diff
@@ -379,6 +379,26 @@ def sendall_with_maxconf(self):
         assert_equal(len(self.wallet.listunspent()), 1)
         assert_equal(self.wallet.listunspent()[0]['confirmations'], 6)
 
+    @cleanup
+    def sendall_spends_unconfirmed_change(self):
+        self.log.info("Test that sendall spends unconfirmed change")
+        self.add_utxos([17])
+        self.wallet.sendtoaddress(self.remainder_target, 10)
+        assert_greater_than(self.wallet.getbalances()["mine"]["trusted"], 6)
+        self.test_sendall_success(sendall_args = [self.remainder_target])
+
+        assert_equal(self.wallet.getbalance(), 0)
+
+    @cleanup
+    def sendall_spends_unconfirmed_inputs_if_specified(self):
+        self.log.info("Test that sendall spends specified unconfirmed inputs")
+        self.def_wallet.sendtoaddress(self.wallet.getnewaddress(), 17)
+        assert_greater_than(self.wallet.getbalances()["mine"]["untrusted_pending"], 0)
```

Comment:
In c9746711:

Need to flush the validation queue before checking the wallet balance. The transaction could have not been processed by the validation interface worker thread. Call `self.wallet.syncwithvalidationinterfacequeue()`.

(saw it failing locally).

Reply:
Also, would be better to check for the exact amount. You are sending exactly 17 btc to the wallet.

Reply:
Done